### DISCUSSION TITLE
How can I use rules to modify an imported field to a variable?

### DISCUSSION THREAD

**Discussion by 2fst4u** [firefly-iii/firefly-iii#6847] [Firefly III] [answered]
For some reason my bank uses a field called "details" as a good option for the opposing account value, except when I use my Visa card, in which case it just says "Visa [redacted card number]" and has a value in a different field called "code".

For almost every transaction I want "details" to be my opposing account, except if its a Visa transaction when I want opposing account to be the "code".

I have mapped "details" to opposing account, and "code" to description, and made a trigger in rules for any transactions where the destination starts with my Visa card text, but I can't figure out how to make the action something like "change the destination to [the value in description]"

Can I use a variable in the action to make this work?

**Comment by JC5** [comment, marked as answer]
This isn't possible, but you could use webhooks instead? The next release of Firefly III will feature a UI for webhooks.

https://docs.firefly-iii.org/firefly-iii/about-firefly-iii/what-its-not/#more-complicated-rules-like-and-or-or-copypasting-values-etc

**Reply by 2fst4u** [reply]
Bother. 

Interesting, how would webhooks work for this?

**Reply by JC5** [reply]
If you have a place where you can process webhooks, like a script, you can push any change back to Firefly III.
